### ISSUE TITLE
Prevent NVDA to read all content when opening a dialog in QT based applications

### ISSUE THREAD

**Issue by Oreonan** [nvaccess/nvda#12420]
### Steps to reproduce:
- Open a QT based application (tested with TeamTalk client)
- Open a dialog of this app (E.G. preferences)

### Actual behavior:
- When opening a dialog, NVDA read absolutly all content of this new dialog, when a dialog contains a lot of content, for example when opening a private message dialog with an user in TeamTalk app, NVDA freeze and we have to wait until we can reoperate with NVDA

### Expected behavior:
- NVDA should be announce only content which has focus, like in all other applications type

#### NVDA installed/portable/running from source:
- Installed

#### NVDA version:
- Always latest alpha

#### Windows version:
- 10.0.19043

#### Name and version of other software in use when reproducing the issue:
- TeamTalk 5.7.1

#### Does the issue still occur after restarting your computer?
- Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
- No

#### If add-ons are disabled, is your problem still occurring?
- Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
- Yes

**Comment by Brian1Gaff** [comment]
When I asked about this odd behaviour I was told it very much depends on how 
it was written, which I'll accept but what is the difference between ones 
that do work and ones that don't for us non clued in types?
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Oreonan" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Friday, May 14, 2021 7:57 PM
Subject: [nvaccess/nvda] Prevent NVDA to read all content when opening a 
dialog in QT based applications (#12420)

**Comment by Oreonan** [comment]
If someone explain how to resolve this problem in the app's code I can 
do this for projects who I'm contributor, but for other apps it's 
problematic.

**Comment by OzancanKaratas** [comment]
This issue should be resolved by NVDA. Because the same problem occurs in all Qt applications. I also tested it on VLC and Format Factory.

**Comment by Oreonan** [comment]
I confirm again there is a NVDA issue, not reproduce this with other screenreaders...

**Comment by CoBC** [comment]
I'm sorry to raise this issue, but if someone can look in to it, it can be really appreciate. I still taking this example, but if you have a long discussion in private with a TeamTalk user for example, wish is use a lot in blind community, when PM dialog gives focus, NVDA freeze and it's really ennoying.
Thanks

**Comment by Adriani90** [comment]
Did you try to create an NVDA profile for the specific application and disable object description in the object presentation settings of NVDA? This might work.

**Comment by CoBC** [comment]
Ah, yes, it's working. Thanks for the workaround but I continue to think 
that a fix for this should be include...

**Comment by michaelDCurran** [comment]
Another work around on the QT app side is to expose a description for the dialog. Specifically, expose UIA fullDescription or helpText on the dialog itself. this will stop NVDA from trying to generate its own dialog text.

this is a complicated issue as many dialogs in windows don't expose their captions in a semantic  way thus requiring NvDA to descend into the dialog, collecting text to make up a possible description. If we were to disable this functionality by default, many dialogs in Windows would no longer have their captions read. 
However, This code is also causing us performance issues in Visual Studio dialogs as well, so it may be time for us to revisit this whole approach.
If we could be sure we could identify the particular dialogs (usually just message boxes) in QT and other frameworks that don't semantically provide a caption yet require us to read a generated one, then we could limit this code to those. But that will be very tricky to identify.
Perhaps for Qt at least, we could disable this for all dialogs with a UIA framework of Qt, except for those with a UIA classname of QMessagebox.
Preferrably though it would be good if QMessagebox exposed its caption on the dialog via UIA fullDescriptionProperty anyway though.
but for backwards compatibility we'd still have to use our code I think.

Someone could certainly experiment doing this in a teamtalk5 appModule, to address it for Teamtalk at very least.

**Comment by josephsl** [comment]
Hi, can we somehow let QT use IsDialog UIA property (IUIA6) at the cost of limiting this to Windows 10 Version 1809/Server 2019 and later? Thanks.

**Comment by CoBC** [comment]
DO you know how to expose an UIA property in a Qt app side? I can 
implement UIA fullDescription for message boxes in TT, but I don't know 
how to expose UIA properties for now :(

**Comment by Adriani90** [comment]
@CoBC is this now fixed in QT 6.7 with NVDA 2024.4.1? Or do you still need guidance?
